### ISSUE TITLE
Suggestion - Incorporate batched Metameric loss

### ISSUE THREAD

**Issue by Gnefil** [kaanaksit/odak#131]
Hii all,

I recently found out that Metameric loss demands the two images to be in the shape of (1, c, h, w), which fails when batch size is larger than one. I wonder if it is possible to add batched operation functionality in Metameric loss, it would be of great convenience. 

Thanks in advance.

**Comment by AlberTgarY** [comment]
sorry for replying late Gnefil, could you try this updated batched version and let me know if things work?
if you could clone this fork 
https://github.com/AlberTgarY/odak/tree/master
and install locally with `pip install -e .`


I used the following code to test: 

```
import torch
import time
from odak.learn.perception import MetamericLoss
def test_metameric_batch():
    batch_sizes = [1, 1, 2, 4, 8]
    device = torch.device('cuda')
    for batch_size in batch_sizes:
        print(f"\n{'='*50}")
        print(f"Testing batch size: {batch_size}")

        image = torch.randn(batch_size, 3, 256, 256, device=device)
        target = torch.randn(batch_size, 3, 256, 256, device=device)
        gaze = [0.5, 0.5]
        
        print(f"Image shape: {list(image.shape)}")
        print(f"Target shape: {list(target.shape)}")
        
        loss_fn = MetamericLoss(
            device=device,
            alpha=0.2,
            n_pyramid_levels=3,  
            use_l2_foveal_loss=True,
            fovea_weight=20.0
        ).to(device)
        
        start_time = time.time()
        
        loss = loss_fn(image, target, gaze=gaze)
        end_time = time.time()
        
        print(f"Loss computed successfully: {loss.item():.6f}")
        print(f"Processing time: {(end_time - start_time):.3f}s")
        print(f"Loss shape: {list(loss.shape) if loss.dim() > 0 else 'scalar'}")

        if image.requires_grad:
            loss.backward()
            print("Gradients successfully")
    print(f"\n{'='*50}")
    print(" ALL TESTS PASSED")
    return True

if __name__ == "__main__":
    # Run the test
    test_metameric_batch()
```



it returns :

> WIN11@LondonUnderdog MINGW64 /d/ALBERT/UCL/odak (master)
> $ python test_meta.py
> Testing on device: cuda
> 
> ==================================================
> Testing batch size: 1
> Image shape: [1, 3, 256, 256]
> Target shape: [1, 3, 256, 256]
> Loss computed successfully: 4.443471
> Processing time: 0.226s
> Loss shape: scalar
> 
> ==================================================
> Testing batch size: 1
> Image shape: [1, 3, 256, 256]
> Target shape: [1, 3, 256, 256]
> Loss computed successfully: 4.473997
> Processing time: 0.070s
> Loss shape: scalar
> 
> ==================================================
> Testing batch size: 2
> Image shape: [2, 3, 256, 256]
> Target shape: [2, 3, 256, 256]
> Loss computed successfully: 4.497545
> Processing time: 0.070s
> Loss shape: scalar
> 
> ==================================================
> Testing batch size: 4
> Image shape: [4, 3, 256, 256]
> Target shape: [4, 3, 256, 256]
> Loss computed successfully: 4.509216
> Processing time: 0.072s
> Loss shape: scalar
> 
> ==================================================
> Testing batch size: 8
> Image shape: [8, 3, 256, 256]
> Target shape: [8, 3, 256, 256]
> Loss computed successfully: 4.489837
> Processing time: 0.077s
> Loss shape: scalar
> 
> ==================================================
>  ALL BATCH TESTS PASSED
> (base)
> 




if it works, I will update the unit test and create the pull request.

**Comment by Gnefil** [comment]
Hii @AlberTgarY , thanks for your contribution. I have tried the new version locally, and I confirm it runs well. What's more, after experimenting, it's a great speed-up compared to summing and averaging images separately. 

Can I ask when this v0.2.7 is going to be released? Thanks again!

**Comment by kaanaksit** [comment]
Please do let us know @Gnefil if this recent commit from @AlberTgarY has resolved the issue. I am closing the issue for now. But please feel free to open it if you think the results are not as intended.
